velox/exec/AssignUniqueId.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/AssignUniqueId.h"
+
+#include <utility>
+
+namespace facebook::velox::exec {
+
+AssignUniqueId::AssignUniqueId(
+    int32_t operatorId,
+    DriverCtx* driverCtx,
+    const std::shared_ptr<const core::AssignUniqueIdNode>& planNode,
+    int32_t uniqueTaskId,
+    std::shared_ptr<std::atomic_int64_t> rowIdPool)
+    : Operator(
+          driverCtx,
+          planNode->outputType(),
+          operatorId,
+          planNode->id(),
+          "AssignUniqueId") {
+  isIdentityProjection_ = true;
```

Comment:
This flag means that all columns are projects as-is without any modification. This would be the case for filter. This operator uses identity projection for all, but one column (id), hence, this flag should be false.

You may want to populate identityProjections_ for all the input columns, use results_ vector to hold the generated "id" column and populate resultProjections_. Then use fillOutput() method from the base class.